docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1282 @@
+and the corresponding method in `ReflectionManager` is invoked.
+
+### Habit tracker component
+
+#### Description
+The habit tracker component aims to provide user with a tool to track and cultivate good habits.
+
+#### Design Considerations
+* ##### User Design Considerations
+    * User can add new habits in the habit tracker that they wish to cultivate.
+    * User can increase the count of a habit after they completed the habit for the day, allowing them to keep track of the total number of times they completed the habit.
+    * User can also delete the habit if they no longer want to cultivate that habit.
+    * User are able to set the priority of the habits (HIGH, MED, LOW), to prioritise their time on more important habits.
+    * User can also sort the habits according to their priority for better visualization.
+    * A help menu is also provided for users to guide them on how to use the habit tracker.
+    * Error messages with guidance messages will be printed to console if command input by user is invalid.
+
+* #### Developer Design Considerations
+    * Modularity: Encapsulate related functionalities within classes to promote re-usability and maintainability.
+    * Exception Handling: Use of custom exceptions to differentiate between various types of errors, and to handle them appropriately with error messages
+    * Data Encapsulation: Control the access to a class's internal attributes, accessible only through getters and setters
+    * Interface Segregation: Segregation of command interface to represent different command types for specific use cases.
+    * Readability and Maintainability: Descriptive naming, robust documentation for code clarity.
+
+#### Implementation
+
+##### Class Diagram
+![HabitClassDiagram.png](diagrams/habit/HabitClassDiagram.png)
+
+* `HabitTracker` class
+    * Overview
+        * The `HabitTracker` class manages the habit tracker list which contains the habits.
+    * Attributes:
+        * `habitList`: Private attribute representing the list of habits. It is an ArrayList of Habit objects.
+    * Methods:
+        * `HabitTracker()`: Constructs a HabitTracker object. 
+        * `getNumberOfHabits()`: Returns the number of habits in the habitList.
+        * `addHabit(Habit newHabit)`: Adds a new Habit object into the habitList.
+        * `listHabits()`: Prints a list of all habits in habitList .
+        * `isValidHabitID(int habitID)`: Check if a habit ID is valid by comparing with the size of habitList.
+        * `updateHabitCount(int habitID, String updatedCount)`: Update the habit count of a habit.
+        * `deleteHabit(int habitID)`: Delete a habit from habitList.
+        * `setPriorityLevel(int habitID, String priority)`: Set the priority of a habit.
+        * `sortHabits()`: Sort the habits in habitList according to the habits' priorities.
+        * `clearHabitList()`: Delete all habits from habitList.
```

Comment:
Could consider leaving out listing methods where the method names are quite self-explanatory